File: samples/feed-demo-android/app/src/main/java/com/sellwild/sample/SampleIds.kt
package com.sellwild.sample

/**
 * Stable element ids for the Maestro flows (e2e/maestro). The one list is
 * contracts/e2e/ids.json: add an id there first, and never rename one.
 *
 * The app sets them with Modifier.testTag. SampleApp turns on testTagsAsResourceId,
 * so UI Automator (and Maestro) reads each tag as the element's resource-id.
 */
object SampleIds {
    // Tabs
    const val TAB_FEED = "sw.tab.feed"
    const val TAB_ADS = "sw.tab.ads"
    const val TAB_LISTINGS = "sw.tab.listings"
    const val TAB_DIAGNOSTICS = "sw.tab.diagnostics"

    // Feed. The SDK feed sets sw.listing.card and sw.feed.ad on its own rows.
    const val FEED_LIST = "sw.feed.list"
    const val FEED_STATUS = "sw.feed.status"

    // Ads
    const val AD_BANNER = "sw.ad.banner"
    const val AD_BANNER_SIZE = "sw.ad.banner.size"
    const val AD_MREC = "sw.ad.mrec"
    const val AD_MREC_SIZE = "sw.ad.mrec.size"
    const val AD_NATIVE = "sw.ad.native"
    const val AD_NATIVE_SIZE = "sw.ad.native.size"

    // Listings
    const val LISTINGS_LIST = "sw.listings.list"
    const val LISTINGS_REFRESH = "sw.listings.refresh"
    const val LISTINGS_STATUS = "sw.listings.status"
    const val LISTING_CARD = "sw.listing.card"

    // Diagnostics
    const val DIAG_SDK_VERSION = "sw.diag.sdk_version"
    const val DIAG_PARTNER = "sw.diag.partner"
    const val DIAG_CONFIG_SOURCE = "sw.diag.config_source"
    const val DIAG_LISTINGS_URL = "sw.diag.listings_url"
    const val DIAG_FAILURES = "sw.diag.failures"
    const val DIAG_FAILURE_CONTEXT = "sw.diag.failure_context"
}

File: samples/feed-demo-ios/SellwildSample/SampleIDs.swift
/// Stable element ids for the Maestro flows (e2e/maestro). The one list is
/// contracts/e2e/ids.json: add an id there first, and never rename one.
enum SampleID {
    // Tabs
    static let tabFeed = "sw.tab.feed"
    static let tabAds = "sw.tab.ads"
    static let tabListings = "sw.tab.listings"
    static let tabDiagnostics = "sw.tab.diagnostics"

    // Feed
    static let feedList = "sw.feed.list"
    static let feedStatus = "sw.feed.status"

    // Ads
    static let adBanner = "sw.ad.banner"
    static let adBannerSize = "sw.ad.banner.size"
    static let adMrec = "sw.ad.mrec"
    static let adMrecSize = "sw.ad.mrec.size"
    static let adNative = "sw.ad.native"
    static let adNativeSize = "sw.ad.native.size"

    // Listings
    static let listingsList = "sw.listings.list"
    static let listingsRefresh = "sw.listings.refresh"
    static let listingsStatus = "sw.listings.status"
    static let listingCard = "sw.listing.card"

    // Diagnostics
    static let diagSdkVersion = "sw.diag.sdk_version"
    static let diagPartner = "sw.diag.partner"
    static let diagConfigSource = "sw.diag.config_source"
    static let diagListingsUrl = "sw.diag.listings_url"
    static let diagFailures = "sw.diag.failures"
    static let diagFailureContext = "sw.diag.failure_context"
}
